build.gradle

```diff
@@ -4,13 +4,29 @@ plugins {
   id "java"
   id "com.diffplug.spotless" apply false
   id "nebula.release"
+  id "io.github.gradle-nexus.publish-plugin"
 }
 
 release {
   defaultVersionStrategy = Strategies.SNAPSHOT
 }
 
-group = "com.splunk.public"
+nebulaRelease {
+  addReleaseBranchPattern(/v\d+\.\d+\.x/)
+}
+
+nexusPublishing {
+  packageGroup = "com.splunk"
+
+  repositories {
+    sonatype {
+      username = System.getenv('SONATYPE_USERNAME')
+      password = System.getenv('SONATYPE_PASSWORD')
+    }
+  }
+}
+
+group = "com.splunk"
```

Comment:
For some reason this looks like duplication with changes in `agent/build.gradle`. Why do we need both?

Reply:
Hmm, the project group seems to be empty for `:agent`, no idea why. I've decided to just duplicate that for the time being, before I figure out why.

Reply:
Sorry, I got confused actually :) Over there is configuring publishing as in "what jars and pom" and signing, over here is configuring publishing as in "where to push". My mistake, sorry.